Beistellung_Implementierung/system/Coding/src/app/State/DisplayCountdown.c

```diff
@@ -0,0 +1,76 @@
+/***************************************************************************************************
+  (c) NewTec GmbH 2024   -   www.newtec.de
+***************************************************************************************************/
+/**
+ * @file       DisplayCountdown.c
+ *
+ *    Module handles displaying the countdown on the OLED-display 
+ */
+/**************************************************************************************************/
+
+/* INCLUDES ***************************************************************************************/
+#include <stdio.h>
+
+#include "DisplayCountdown.h"
+#include "os/SoftTimer.h"
+#include "service/Display.h"
+
+/* CONSTANTS **************************************************************************************/
+#define TIME_MAX_MS (3000U)
+
+/* MACROS *****************************************************************************************/
+
+/* TYPES ******************************************************************************************/
+
+/* PROTOTYPES *************************************************************************************/
+
+/* VARIABLES **************************************************************************************/
+static SoftTimer gCountdownTimer;
+static UInt8 gLastTime;
+
+/* EXTERNAL FUNCTIONS *****************************************************************************/
+
+void DisplayCountdown_startCountdown(void)
+{
+    //Todo: No proper error handling
+    gLastTime = 0U;
+    SoftTimer_init(&gCountdownTimer);
+    SoftTimerHandler_register(&gCountdownTimer);
+    SoftTimer_start(&gCountdownTimer, TIME_MAX_MS);
+}
+
+EventEnum DisplayCountdown_displayCountdown(void)
+{
+    UInt16 time_ms = SoftTimer_get(&gCountdownTimer);
+    UInt8  time_s = (time_ms + 999U) / 1000U; /* round up to full seconds */
```

Comment:
Magic numbers

1000u could be a define like MsPerSec and to get rid of 999u you could use something like the following function:
```
result = time_ms / MsPerSec;
if (0 != (time_ms % MsPerSec))
{
    ++result;
}
```